flexmeasures/api/v2_0/tests/test_api_v2_0_sensors_fresh_db.py

```diff
@@ -15,8 +15,8 @@
 @pytest.mark.parametrize(
     "post_message",
     [
-        message_for_post_price_data(),
-        message_for_post_price_data(prior_instead_of_horizon=True),
+        message_for_post_price_data(market_id=7),
```

Comment:
@Flix6x please take a look here. I had to use two different market IDs to get this to work, so it seems that when it runs the first time, the db is actually not fresh, but still has some other sensors in it. It might be that this is only being uncovered by using the fm1 scheme which is ID-based and not name-based.

Reply:
I also ran into this messing around with test sensor ids. It is just not future proof. For one thing, test sensor ids are likely to change whenever we set up a new sensor in some conftest.

I had the idea of writing a util function for tests to look up a sensor id by the sensor (display) name. For production sensors those names might change (due to user interaction), but for test sensors they can be considered constants (or at least much more constant than the ids).

Does that idea help you out here?

Reply:
Some options to consider:

- make it possible for conftests to ask to get a specific id when setting up e.g. markets, or
- use something like `message_for_post_price_data(market_id="EPEX day-ahead market")` and have the test call a util function to replace string-type market ids with the actual id of the named market.

Reply:
Thanks for those ideas. I'll pick one.

However, the main issue I'm seeing here is that the fresh database isn't fresh. The same test function code sees different ideas. The first run sees that market as the seventh sensor that was added, and the next parametrised run gets it as the first.

Maybe we need some initial cleaning in the fresh_db fixture?

Reply:
I believe I found the underlying problem in this case. It has partly to do with fixture scope and with the autouse feature.

The current example is in the api 2.0 tests: the `setup_api_test_data` fixture (autouse=True, so all tests request it, also the ones for fresh data) has module scope and pytest runs it first (adding `-s` and a debug print in `fresh_db` confirms this), *before* the `fresh_db` of the first of our two test invocations, which has a lower scope. This fixture adds some assets in the initial run which our "fresh" test doesn' t actually expect. The first invocation thus needs a higher ID. Our `db` fixture cleans up *after* itself, so the second test can work with the expected IDs.

Probably we should not make `setup_api_test_data` autouse=True. If there are two different test data setups, each test needs to specify which test data setup it requires.

Then tests will work if the IDs are the right ones (right order). I believe it owuld be nice to have a more explicit way to assign IDs (like your ideas) but for now I simply want to get this PR mergeable. I'll try out my idea tomorrow.

Reply:
Great that you found out. Less autousing fixtures is generally a good idea, so I am happy with that solution.

Reply:
I'll do this in a separate PR, still within 0.5.0